src/common.ts

```diff
@@ -222,9 +224,13 @@ export interface ArIOWriteContract {
 
 export type WriteInteractionResult = Transaction | DataItem;
 
-export type JoinNetworkParams = GatewayConnectionSettings &
-  GatewayStakingSettings &
-  GatewayMetadata & { qty: number };
+// Utility type for dev experience: shows all fields of a type rather than the composed type names
+export type VisibleTypes<T> = { [P in keyof T]: T[P] };
```

Comment:
This consumes an intersecting type and makes it visible to intellisense - it should rebuild the exact type.

Reply:
I don't love these bc they are 1. hard to read/understand and bc of that 2. hard to change with confidence

I'll offer a commit with what I think might align with what we are trying to acheive.

Reply:
I get what you mean, and im not a fan either - this type serves . Not trying to be pedantic but does this improve readability at all? 
```typescript
export type ShowGranularPropertyTypes<ObjectToShowGranularly> = { [PropertyToShow in keyof ObjectToShowGranularly]: ObjectToShowGranularly[PropertyToShow] };
```